app/Http/Controllers/Maps/DeviceDependencyController.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * DependencyController.php
+ *
+ * Controller for graphing Relationships
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @package    LibreNMS
+ * @link       http://librenms.org
+ * @copyright  2019 Thomas Berberich
+ * @author     Thomas Berberich <sourcehhdoctor@gmail.com>
+ */
+
+namespace App\Http\Controllers\Maps;
+
+use App\Models\Device;
+use Illuminate\Http\Request;
+use LibreNMS\Util\Url;
+
+class DeviceDependencyController extends MapController
+{
+    // Device Dependency Map
+    public function dependencyMap(Request $request)
+    {
+        $devices = Device::hasAccess($request->user())->with('parents')->get();
+
+        $dependencies = [];
+        $devices_by_id  = [];
+
+        // Build the style variables we need
+
+        // List all devices
+        foreach ($devices as $device) {
+            if ($device['disabled']) {
+                $device_style = $this->nodeDisabledStyle();
+            } elseif (! $device['status']) {
+                $device_style = $this->nodeDownStyle();
+            } else {
+                $device_style = $this->nodeUpStyle();
+            }
+
+            // List all Device
+            $devices_by_id[$device['device_id']] = array_merge(
```

Comment:
Why do you need to set array key to device id when you remove it with array_values?

Reply:
It's an adaption of `Map->Network`.
I try it without, run's also fine. So i removed array key setting.